### ISSUE TITLE
Select does not scroll with keyboard input

### ISSUE THREAD

**Issue by benediktgrether** [marigold-ui/marigold#2693]
## Description

We have limited the height of the select so that there are no issues if the select does not have enough space above or below. Now the select is no longer scrolled when the keyboard focus is inside the selected.

## How to reproduce

1. Add following code into Selected Stories

```tsx
export const SelectedScroll = () => {
  return (
    <Select disabledKeys={['Firefly']}>
      <Select.Option key="Harry Potter">Harry Potter</Select.Option>
      <Select.Option key="Lord of the Rings">Lord of the Rings</Select.Option>
      <Select.Option key="Star Wars">Star Wars</Select.Option>
      <Select.Option key="Star Trek">Star Trek</Select.Option>
      <Select.Option key="Firefly-1">Firefly</Select.Option>
      <Select.Option key="Firefly-2">Firefly</Select.Option>
      <Select.Option key="Firefly-3">Firefly</Select.Option>
      <Select.Option key="Firefly-4">Firefly</Select.Option>
      <Select.Option key="Firefly-5">Firefly</Select.Option>
      <Select.Option key="Firefly-6">Firefly</Select.Option>
      <Select.Option key="Firefly-7">Firefly</Select.Option>
      <Select.Option key="Firefly-8">Firefly</Select.Option>
      <Select.Option key="Firefly-9">Firefly</Select.Option>
      <Select.Option key="Firefly-10">Firefly</Select.Option>
      <Select.Option key="Firefly-11">Firefly</Select.Option>
      <Select.Option key="Firefly-12">Firefly</Select.Option>
      <Select.Option key="Firefly-13">Firefly</Select.Option>
      <Select.Option key="Firefly-14">Firefly</Select.Option>
      <Select.Option key="Firefly-15">Firefly</Select.Option>
      <Select.Option key="Firefly-16">Firefly</Select.Option>
      <Select.Option key="Firefly-17">Firefly</Select.Option>
      <Select.Option key="Firefly-18">Firefly</Select.Option>
      <Select.Option key="Firefly-19">Firefly</Select.Option>
      <Select.Option key="Firefly-20">Firefly</Select.Option>
      <Select.Option key="Firefly-21">Firefly</Select.Option>
    </Select>
  );
};
```

2. Got to Storybook  Selected and Selected Scroll.
3. Go with the Keyboard into the Selected. 
4. Try to go down with your arrow keys.
5. Selected doesn't scroll with the arrow keys.

## Expected behavior

The select should scroll with the arrow keys when the next element is no longer in the viewport and bring it to the top.
You can test it here. https://react-spectrum.adobe.com/react-aria/useSelect.html#asynchronous-loading
